File: src/main/java/com/web/rpg/service/monster/MonsterService.java
package com.web.rpg.service.monster;

import com.web.rpg.model.Characters.PlayerCharacter;
import com.web.rpg.model.Monsters.Monster;

public interface MonsterService {

    void applyDamage(Monster monster, double damage);

    Monster prepearMonsterForBattle(PlayerCharacter character);

    void remove(Monster monster);

    Monster updateOrCreate(Monster monster);
}

File: src/main/java/com/web/rpg/service/monster/MonsterService.scala
package com.web.rpg.service.monster

import com.web.rpg.model.Characters.PlayerCharacter
import com.web.rpg.model.Monsters.Monster

trait MonsterService {

  def applyDamage(monster: Monster, damage: Double): Unit

  def prepareMonsterForBattle(character: PlayerCharacter): Monster

  def remove(monster: Monster): Unit

  def updateOrCreate(monster: Monster): Monster

}
